controllers/items.js

```diff
@@ -0,0 +1,151 @@
+const ItemModel = require("../models/items");
+
+/**
+ * Adds a tag to an item with its corresponding a level
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const addTagsToItem = async (req, res) => {
+  try {
+    const { itemid } = await ItemModel.addTagsToItem(req.body);
+    return res.json({
+      message: "Tags added successfully!",
+      itemid,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Removes a tag from an items
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const removeTagsFromItem = async (req, res) => {
+  try {
+    const { itemid, tagid } = await ItemModel.removeTagsFromItem(req.body);
+    return res.json({
+      message: "Tags removed successfully!",
+      itemid,
+      tagid,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items based on a particular item type
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnType = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnType(req.params.type);
+    return res.json({
+      message: "Items fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items based on a particular tag id
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnTagId = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnTagId(req.params.tagid);
+    return res.json({
+      message: "Items fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items with its tags and levels data based on a particular itemid
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnitemId = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnItemId(req.params.itemid);
+    return res.json({
+      message: "Item fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
```

Comment:
Could these be merged into just function, and we could pass an object as argument to select the `itemId` or the `tagtype`?
```js
ItemModel.getItems({ itemId });
ItemModel.getItems({ tagType });

```

Having to maintain a lot of code per field type seems like something we can avoid.